openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py

```diff
@@ -88,39 +97,75 @@ def tearDown(self):
                 if e.errno not in (errno.ENOENT, errno.EACCES, errno.EPERM):
                     raise e
 
-    def test_model_viewer_has_correct_data_from_problem(self):
-        """
-        Verify that the correct model structure data exists when stored as compared
-        to the expected structure, using the SellarStateConnection model.
-        """
-        p = Problem(model=SellarStateConnection())
-        p.setup()
-
-        model_viewer_data = _get_viewer_data(p)
+    def check_model_viewer_data(self, model_viewer_data,
+                                expected_tree,
+                                expected_pathnames,
+                                expected_conns,
+                                expected_abs2prom,
+                                expected_declare_partials,
+                                expected_driver_name,
+                                expected_design_vars_names,
+                                expected_responses_names
+                                ):
 
-        # check expected model tree
-        self.assertDictEqual(model_viewer_data['tree'], self.expected_tree)
+        self.assertDictEqual(model_viewer_data['tree'], expected_tree)
 
         # check expected system pathnames
         pathnames = model_viewer_data['sys_pathnames_list']
-        self.assertListEqual(sorted(pathnames), self.expected_pathnames)
+        self.assertListEqual(sorted(pathnames), expected_pathnames)
 
         # check expected connections, after mapping cycle_arrows indices back to pathnames
-        connections = sorted(model_viewer_data['connections_list'], key=lambda x: (x['tgt'], x['src']))
+        connections = sorted(model_viewer_data['connections_list'],
+                             key=lambda x: (x['tgt'], x['src']))
+        expected_conns = sorted(expected_conns,
+                             key=lambda x: (x['tgt'], x['src']))
         for conn in connections:
             if 'cycle_arrows' in conn and conn['cycle_arrows']:
                 cycle_arrows = []
                 for src, tgt in conn['cycle_arrows']:
                     cycle_arrows.append(' '.join([pathnames[src], pathnames[tgt]]))
                 conn['cycle_arrows'] = sorted(cycle_arrows)
-        self.assertEqual(len(connections), len(self.expected_conns))
-        for c, ex in zip(connections, self.expected_conns):
+            self.assertEqual(len(connections), len(expected_conns))
```

Comment:
looks like an indentation error..  you don't mean to check this in the loop do you?

Reply:
Good catch. Fixed.